### PR TITLE
Pro 1958 will codicil date remove

### PR DESCRIPTION
**Pull request by harjeetkalsi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRO-1958

### Change description ###

Removing will date and codicils date question, which changes the journey and removed from the check your answers page. 

DO NOT MERGE

### CODE REVIEW

File: app/components/submit-data.js [hunk 1]
```diff
@@ -28,9 +28,6 @@ const dataMap = {
     willWithDate: 'will.isWillDate',
     willWithCodicils: 'will.codicils',
     willCodicilsNumber: 'will.codicilsNumber',
-    willCodicilsWithDate: 'will.isCodicilsDate',
-    willCodicilsDate: 'will.codicilsDate_formattedDate',
-    willDate: 'will.willDate_formattedDate',
```
Comment (shepda) [inline app/components/submit-data.js]: We've removed the `isWillDate` field so we could also remove `willWithDate: 'will.isWillDate',`?

File: test/component/summary/testWillSection.js [hunk 2]
```diff
@@ -59,12 +57,10 @@ describe('summary-will-section', () => {
                     const playbackData = {};
                     playbackData.willLeft = willContent.left.question;
                     playbackData.willOriginal = willContent.original.question;
-                    playbackData.date = willContent.date.question;
-                    playbackData.dateQuestion = willContent.date.willDateQuestion;
                     playbackData.willCodicils = willContent.codicils.question;
                     playbackData.applicantExecutor = applicantContent.executor.question;
 
-                    Object.assign(playbackData, willData.will, willData.applicant);
+                    Object.assign(playbackData, willData.applicant);
```
Comment (shepda) [inline test/component/summary/testWillSection.js]: I don't think we want to remove `willData.will`, rather we need to remove the now unused test data from `test/data/will.json`.

File: app/steps/ui/will/codicils/index.js [hunk 3]
```diff
@@ -24,11 +24,6 @@ module.exports = class WillCodicils extends ValidationStep {
         if (ctx.codicils === this.generateContent(ctx, formdata).optionNo) {
             delete ctx.codicilsNumber;
             delete ctx.isCodicilsDate;
-            delete ctx.codicilsDate_day;
-            delete ctx.codicilsDate_month;
-            delete ctx.codicilsDate_year;
-            delete ctx.codicilsDate_date;
-            delete ctx.codicilsDate_formattedDate;
```
Comment (shepda) [inline app/steps/ui/will/codicils/index.js]: Do we also want to remove `delete ctx.isCodicilsDate;`?

File: test/component/deceased/testMarried.js [hunk 4]
```diff
@@ -40,7 +40,7 @@ describe('deceased-married', () => {
                     'firstName': 'Mana', 'lastName': 'Manah'
                  },
                  will: {
-                     isCodicilsDate: 'Yes'
+                     codicils: 'Yes'
```
[the remark this answers is not in the document]
Comment (harjeetkalsi) [reply test/component/deceased/testMarried.js]: yes done

File: app/components/submit-data.js (hunk 1 above)
Comment (harjeetkalsi) [reply app/components/submit-data.js]: yep good spot!

File: app/steps/ui/will/codicils/index.js (hunk 3 above)
Comment (harjeetkalsi) [reply app/steps/ui/will/codicils/index.js]: Yes now done

**Pull request merged**